modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Home/index.js

```diff
@@ -35,76 +38,96 @@ const getStatus = (slaCurrent, slaFuture) => {
 	return STATUS_TAG_TYPES.expired;
 };
 
+const getKoroneikiFilter = (accounts) => {
+	return accounts
+		?.map(
+			({externalReferenceCode}, index, {length: totalAccounts}) =>
+				`accountKey eq '${externalReferenceCode}'${
+					index + 1 < totalAccounts ? ' or' : ''
+				}`
+		)
+		.join(' ');
```

Comment:
I think that .map followed by .join could be an reduce:


```suggestion
		?.reduce((acc, {externalReferenceCode}, index, {length: totalAccounts}) => `${acc}accountKey eq '${externalReferenceCode}'${index + 1 < totalAccounts && ' or '}`, '');
```

And just one iteration will be done. What do you think?

Reply:
I find the change interesting